File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        unordered_map<char, int> mp;
        
        for(int i=0; i<s.length(); i++){
            mp[s[i]]++;
        }
        
        for(int i=0; i<t.length(); i++){
            mp[t[i]]--;
            if(mp[t[i]] == 0) mp.erase(t[i]);
        }
        
        if(mp.size() == 0) return true;
        else return false;
    }
};

File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s) != len(t):
            return False

        freq = {}

        for c in s:
            freq[c] = freq.get(c, 0) + 1

        for c in t:
            if c not in freq:
                return False
            else:
                freq[c] = freq.get(c)-1
                if freq[c] == 0:
                    del freq[c]
        
        return len(freq) == 0